cmd/skopeo/sync.go

```diff
@@ -240,6 +253,75 @@ func syncSourceHandler(c *cli.Context, globalDestRef types.ImageReference) (toCo
 	return
 }
 
+func syncSourceFileHandler(c *cli.Context) (toCopyList []imagesBySource, err error) {
+	var cfg sourceCfg
+	if err = cfg.Unmarshal(c.String("source-file")); err != nil {
+		return nil, fmt.Errorf("Unmarshaling failed: %s", err)
+	}
```

Comment:
This is just a matter of personal taste, but I'm used to see other Go programs define allocator functions. For example I would expect something like that:

```go
cfg, err := newSourceCfg(c.String("source-file"))
```

The `newSourceCfg` would be made from some of the above code and the `Unmarshal` function (which would disappear)